BAND v04

wildsm opened the pull request:
This is a draft PR with staging for the bandframework v0.4 release.


The working checklist is from the updated release process in develop (https://github.com/bandframework/bandframework/blob/develop/resources/dev_guide/release-proc.rst).


BAND Framework Release Process
===============

A release can be undertaken only by a project administrator.
A project administrator should have an administrator role on the `bandframework
GitHub <https://github.com/bandframework>`_.

Best practice is to follow the version of this process as recorded on the ``develop`` and/or release branch(es). 

Before release
--------------

- [x] A GitHub issue is created with a checklist for the release.

  - Working through issue #110 as well as this PR (#131)

- [x] A release branch should be taken off ``develop`` (or ``develop`` pulls
  controlled).

  - `develop` is controlled (through branch rules); it will be the release branch. There is a post-release branch for `develop`; see #143 

- [x] Release notes for this version are added to the `CHANGELOG.rst </CHANGELOG.rst>`_ file.

  - Confirmed

- [x] Ensure that links and references have been updated (e.g., no occurrences of ``privateband``).

  - `grep` shows no unexpected `privateband` occurrences in `develop`:

```
./.github/config/.wordlist.txt:privateband
./resources/dev_guide/release-proc.rst:- Ensure that links and references have been updated (e.g., no occurrences of ``privateband``).
./CHANGELOG.rst:- improve documentation and facilitation of migration from privateband and bandframework branches to bandframework
./CHANGELOG.rst:- improve documentation and facilitation of migration from privateband and bandframework branches to bandframework
./CHANGELOG.rst:- improve documentation and facilitation of migration from privateband and bandframework branches to bandframework
```

- [x] Version number is updated wherever it appears and ``+dev`` suffix is removed
  (in `README.md </README.md>`_ and possibly in `CHANGELOG.rst </CHANGELOG.rst>`_).

  - Confirmed (after reviewer comments in this PR)

- [x] Check `README.md </README.md>`_ *Citing bandframework* for correctness (e.g., ensure that author list matches `AUTHORS </AUTHORS>`_).

  - This is confirmed; all author lists were updated for v0.4.

- [x] Check `bandsdk.md </resources/sdkpolicies/bandsdk.md>`_ *Citing bandframework* for correctness. Ensure that SDK version listed is the same as in the provided `template </resources/sdkpolicies/template.md>`_.

  - Confirmed by reviewers
 
- [x] Check `sdkpolicies README </resources/sdkpolicies/README.md>`_ for correctness (e.g., ensure that each software in the release has a working link for its SDK compliance, also note any differences in the version of the SDK that each package lists versus the version at `bandsdk.md </resources/sdkpolicies/bandsdk.md>`_).

- [x] Tests are run with source to be released (this may iterate):

  - [x] Online CI (GitHub Actions) tests must pass.

  - [x] Documentation must build and display correctly wherever hosted.

- [x] Pull request from either the ``develop`` or release branch to ``main`` requesting
  one or more reviewers (including at least one other administrator).

- [x] Reviewer will check that all tests have passed and will then approve merge.

- [x] Send an email to bandframework@cels.anl.gov asking to confirm whether the list of email contacts in `CODE_OF_CONDUCT.md </CODE_OF_CONDUCT.md>`_ is up-to-date.

  - All 5 individuals listed have confirmed things are working as expected

During release
--------------

An administrator will take the following steps.

- [x] Merge the pull request into ``main``.

- [x] Once CI tests have passed on ``main``:

  - [x] A GitHub release will be taken from the ``main``.

- [x] If the merge was made from a release branch (instead of ``develop``), merge this
  branch into ``develop``.

- [x] Create a new commit on ``develop`` that appends ``+dev`` to the version number
  (wherever it appears).

After release
-------------

- [x] Ensure all relevant GitHub issues are closed.

- [ ] Update website to reflect existence of new release.

- [ ] Disseminate news of new release to: FRIB-TA mailing list, JETSCAPE/X-SCAPE mailing list, other interested parties.

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
not much
Merge remote-tracking branch 'refs/remotes/origin/v03/smoothemulatorscott' into v03/smoothemulatorscott

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

cdrischler committed:
adds 'nuclear_saturation' submodule

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

beykyle committed:
add jitr as submodule, tag to v1.2

beykyle committed:
update readme to reference jitr

beykyle committed:
update to BAND compatible version

Commit:
point to correct jitr version

wildsm committed:
Update README.md

Fixed error in URL (found by the github action in bandframework)

wildsm committed:
Delete resources/workshops/Data2024/README.md

This is not (and won't be) in develop

wildsm committed:
Update .wordlist.txt

Adding jitr

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

wildsm committed:
Merge pull request #107 from bandframework/develop

Bring branch up to date with develop

wildsm committed:
Update README.md

To address a spelling issue

wildsm committed:
Update readme.md

wildsm committed:
Merge pull request #108 from bandframework/wildsm-patch-2

Wildsm patch 2

wildsm committed:
Update readme.md

wildsm committed:
Merge pull request #109 from bandframework/wildsm-patch-2

Update readme.md

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
huh?
Merge remote-tracking branch 'refs/remotes/origin/v03/smoothemulatorscott' into v03/smoothemulatorscott

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

wildsm committed:
Update .gitmodules

Attempts to fix the warning

`
warning: redirecting to https://github.com/asemposki/SAMBA.git/
`

that I receive when doing a fresh recursive clone.

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
yada yada

outra-banda committed:
Update README.md

wildsm committed:
Update .wordlist.txt

For the updated FrescoX

wildsm committed:
Delete resources/workshops/Data2024 directory

Making consistent with develop branch

wildsm committed:
Merge pull request #115 from bandframework/outra-banda-patch-1

Update README.md

scottedwardpratt committed:
yada yada

wildsm committed:
Update release-proc.rst

wildsm committed:
Update release-proc.rst

wildsm committed:
Update release-proc.rst

mosesyhc committed:
Merge pull request #116 from bandframework/113-ensure-sdk-list-is-updated-prior-to-release-v04

113 ensure sdk list is updated prior to release v04. 
Reviewed addition to `release-proc.rst`.

beykyle committed:
point to v1.4 jitr

wildsm committed:
Update .gitmodules

wildsm committed:
Update parmoo-bandsdk.md

New email

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update .wordlist.txt

wildsm committed:
Update README.md

wildsm committed:
Update .wordlist.txt

wildsm committed:
Merge pull request #111 from bandframework/wildsm-patch-2

Update .gitmodules

wildsm committed:
Update README.md

asemposki committed:
updating the submodule SAMBA

ozgesurer committed:
Adding PUQ sdk

ozgesurer committed:
update SDK policies readme

ozgesurer committed:
Add sdk

ozgesurer committed:
Update BAND Readme

ozgesurer committed:
Update wordlist to include PUQ

ozgesurer committed:
Removing my email's link

wildsm committed:
Update PUQ-bandsdk.md

wildsm committed:
Update README.md

We seem to be alphabetical order

wildsm committed:
Update README.md

wildsm committed:
Update README.md

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update README.md

wildsm committed:
Update README.md

wildsm committed:
Merge pull request #117 from bandframework/release/parmoo_v0.4

Update parmoo-bandsdk.md

wildsm committed:
Merge branch 'v04/PUQ' into develop

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Temporary URL fix for actions

debugging...

wildsm committed:
Update bandframework-ci.yml

wildsm committed:
Update README.md

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update README.md

wildsm committed:
Update README.md

Found a missing submodule

wildsm committed:
Merge pull request #118 from asemposki/develop

updating the submodule SAMBA

wildsm committed:
Merge pull request #121 from bandframework/patch/CHANGELOGv04-PUQ

Update CHANGELOG.rst

wildsm committed:
Merge pull request #122 from bandframework/develop

Bringing the latest develop into smoothemulator

Commit:
resolve merge conflicts

kylegodbey committed:
update rose repo

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update README.md

wildsm committed:
Update CHANGELOG.rst

scottedwardpratt committed:
yada yada

scottedwardpratt committed:
Merge remote-tracking branch 'refs/remotes/origin/v03/smoothemulatorscott' into v03/smoothemulatorscott

cdrischler committed:
Updates submodule nuclear_saturation to the latest commit

mosesyhc committed:
surmise v0.3.0

mosesyhc committed:
spellcheck wordlist update

mosesyhc committed:
Merge branch 'develop' into v04/surmise

mosesyhc committed:
Revert "spellcheck wordlist update"

This reverts commit 470cb13d7fc04cd4f2f8e27b967eb4fadbfa2333.

mosesyhc committed:
Reapply "spellcheck wordlist update"

This reverts commit 00582e12315b65a2b43d753b42fd7298459a3625.

mosesyhc committed:
Merge branch 'v04/surmise' of https://github.com/bandframework/bandframework into v04/surmise

mosesyhc committed:
reverting spellcheck wordlist for removed document.

wildsm committed:
Added nsat sdk to index

wildsm committed:
Update bandframework-ci.yml

wildsm committed:
Update bandframework-ci.yml

wildsm committed:
Merge pull request #126 from bandframework/develop

Bringing the latest develop into CD branch

wildsm committed:
Update bandframework-ci.yml

Including nsat-bandsdk.md

wildsm committed:
Update .wordlist.txt

Adding nsat

wildsm committed:
Merge pull request #127 from bandframework/main

bringing dev up to date with main

wildsm committed:
Update README.md

cdrischler committed:
updates nuclear saturation after feedback from S. Wild

wildsm committed:
Update CHANGELOG.rst

added nsat to changelog

wildsm committed:
Merge pull request #124 from bandframework/v04/christian

Merge V04/christian into develop requested

wildsm committed:
Update .wordlist.txt

wildsm committed:
Temp update of .wordlist.txt

wildsm committed:
Merge pull request #128 from bandframework/develop

bringing surmise up to date with develop

wildsm committed:
Undoing of temp update of .wordlist.txt

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update README.md

scottedwardpratt committed:
yada yada

kylegodbey committed:
Update CHANGELOG.rst

scottedwardpratt committed:
yada yada

mosesyhc committed:
surmise 0.3.0 reupdate

wildsm committed:
Update template.md

Cleanup

asemposki committed:
updating Taweret submodule

asemposki committed:
Merge pull request #129 from asemposki/v04/taweret

updating Taweret submodule

mosesyhc committed:
Merge pull request #125 from bandframework/v04/surmise

Including surmise v0.3.0 to BAND framework

kylegodbey committed:
Update rose submodule to latest

wildsm committed:
Merge pull request #123 from bandframework/v04/rose

Update ROSE to v1.1.3

wildsm committed:
Merge branch 'develop' into main

scottedwardpratt committed:
yada yada

wildsm committed:
Update SmoothEmulatorSDK.md

Minor edits to sdk file.

wildsm committed:
Update README.md

wildsm committed:
Update README.md

wildsm committed:
Merge pull request #132 from bandframework/develop

Bringing up to date with develop

wildsm committed:
Update README.md

wildsm committed:
Added to SDK Index

Added to SDK Index

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update .wordlist.txt

wildsm committed:
Update bandframework-ci.yml

wildsm committed:
Update README.md

Nice, spellcheck worked

wildsm committed:
Merge pull request #72 from bandframework/v03/smoothemulatorscott

V03/smoothemulatorscott

wildsm committed:
Adding a README in BAND for PUQ

ozgesurer committed:
Replacing parmoo with PUQ

ozgesurer committed:
Updating README with a tutorial

ozgesurer committed:
Update wordlist and README

ozgesurer committed:
Fix in README and wordlist

ozgesurer committed:
Merge pull request #134 from bandframework/v04/PUQadditional

Adding a README in BAND for PUQ

Commit:
point to v2.0

Commit:
update to v2.0

Commit:
Merge branch 'main' of github.com:beykyle/bandframework

Commit:
merge

asemposki committed:
updating submodule Taweret

asemposki committed:
Merge pull request #135 from asemposki/v04/taweret

updating submodule Taweret for v0.4

Commit:
update to v2.0.1

wildsm committed:
Typo in hyperlink markup

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Merge branch 'develop' into main

wildsm committed:
Add sdk link

wildsm committed:
Update .wordlist.txt

wildsm committed:
Update bandframework-ci.yml

wildsm committed:
Merge pull request #103 from beykyle/main

Add jitr as BAND software for v04

wildsm committed:
Merge pull request #136 from bandframework/develop

Bringing up to date with develop

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update README.md

asemposki committed:
Merge pull request #130 from bandframework/v04/taweret

Updating Taweret for v0.4

wildsm committed:
Added links to changelog

wildsm committed:
Update CONTRIBUTING.rst

wildsm commented on .github/config/.wordlist.txt:
```diff
@@ -45,36 +111,21 @@ directbayes
 frontend
 functionals
-Furnstahl
+gKTCLEAGE
```
@mosesyhc - does this need to be in the wordlist (in either band or surmise)?

wildsm replied:
Great. Actions succeed after removing from the BAND wordlist: https://github.com/bandframework/bandframework/pull/138/commits/f1b2b94c9a7ac772db36653f9b6f85ed5cdeae0e

wildsm committed:
Update release-proc.rst

wildsm committed:
Update release-proc.rst

wildsm committed:
Update README.md

Caught a typo and trying to get individual packages recognized.

wildsm committed:
Update bandsdk.md

wildsm commented on software/SmoothEmulator/doc/files/UserManual.tex:
```diff
@@ -0,0 +1,199 @@
+       
+       \begin{center}
+     
+       \includegraphics[width=0.24\textwidth]{figs/BAND_logo.png}
```
What do people prefer: 
- for the SmoothEmulator documentation to be entirely self contained (so keep the separate png files)
- use the existing logo (so link to ../../../../resources/BAND_logo_v2.png)
- something else?

wildsm commented on software/SmoothEmulator/doc/files/figs/nsf_logo.jpg:
Are both the png and jpg nsf logos needed?
@scottedwardpratt

wildsm replied:
Scott replied below

wildsm committed:
Update AUTHORS v0.4

Waiting for an additional confirmation

wildsm committed:
Update README.md

Updated v0.4 authors

wildsm committed:
Update bandsdk.md

Updating authors

scottedwardpratt commented:
Wow, now that’s thorough. No, latex uses the png. We can get rid of the jpeg. I downloaded the jpeg from NSF, then converted it to png for latex.

There are some other pieces of dead code (functional perhaps, but not listed in the manual and not really tested) and I don’t see any harm in keeping it around. E.g., we have some pybind11 code that works, but is still being given some added functionality. The pybind11 will be incorporated into the next release (this allows the emulator to work within a python notebook, so it can work fairly seamlessly, not that anything with pybind11 is ever robust, with surmise).

Thanks so much again — Scott

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Scott Pratt                                        ***@***.*** %
% FRIB                                               +1-517-908-7460 %
% Michigan State University                            Office   2133 %
% East Lansing, MI 48824    http://nscl.msu.edu/directory/pratt.html %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%




















On Sep 24, 2024, at 10:32 PM, Stefan M Wild ***@***.***> wrote:



@wildsm commented on this pull request.

________________________________

On software/SmoothEmulator/doc/files/figs/nsf_logo.jpg<https://urldefense.com/v3/__https://github.com/bandframework/bandframework/pull/131*discussion_r1774321968__;Iw!!HXCxUKc!10Dm_i8a1QmXyxD1If94ZGFTN60rZMdQVOOsXs9TzhsKjtfz6ON5gHVqa0kdVAg69lDzPBDqq3l7w9lSugKexV4$>:

Are both the png and jpg nsf logos needed?
@scottedwardpratt<https://urldefense.com/v3/__https://github.com/scottedwardpratt__;!!HXCxUKc!10Dm_i8a1QmXyxD1If94ZGFTN60rZMdQVOOsXs9TzhsKjtfz6ON5gHVqa0kdVAg69lDzPBDqq3l7w9lSfZFT-h4$>

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/bandframework/bandframework/pull/131*pullrequestreview-2326841388__;Iw!!HXCxUKc!10Dm_i8a1QmXyxD1If94ZGFTN60rZMdQVOOsXs9TzhsKjtfz6ON5gHVqa0kdVAg69lDzPBDqq3l7w9lS7Ae740g$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AB6P62WFNP3E4CBERN6UWYLZYIOCZAVCNFSM6AAAAABOOAHBRGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGMRWHA2DCMZYHA__;!!HXCxUKc!10Dm_i8a1QmXyxD1If94ZGFTN60rZMdQVOOsXs9TzhsKjtfz6ON5gHVqa0kdVAg69lDzPBDqq3l7w9lS19nNY94$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

wildsm committed:
Delete extra NSF logo

wildsm commented:
> Wow, now that’s thorough. No, latex uses the png. We can get rid of the jpeg. I downloaded the jpeg from NSF, then converted it to png for latex. There are some other pieces of dead code (functional perhaps, but not listed in the manual and not really tested) and I don’t see any harm in keeping it around. E.g., we have some pybind11 code that works, but is still being given some added functionality. The pybind11 will be incorporated into the next release (this allows the emulator to work within a python notebook, so it can work fairly seamlessly, not that anything with pybind11 is ever robust, with surmise). Thanks so much again — Scott %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% % Scott Pratt ***@***.*** % % FRIB +1-517-908-7460 % % Michigan State University Office 2133 % % East Lansing, MI 48824 http://nscl.msu.edu/directory/pratt.html % %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% On Sep 24, 2024, at 10:32 PM, Stefan M Wild ***@***.***> wrote: @wildsm commented on this pull request.
> […](#)
> ________________________________ On software/SmoothEmulator/doc/files/figs/nsf_logo.jpg<https://urldefense.com/v3/__https://github.com/bandframework/bandframework/pull/131*discussion_r1774321968__;Iw!!HXCxUKc!10Dm_i8a1QmXyxD1If94ZGFTN60rZMdQVOOsXs9TzhsKjtfz6ON5gHVqa0kdVAg69lDzPBDqq3l7w9lSugKexV4$>: Are both the png and jpg nsf logos needed? @scottedwardpratt<https://urldefense.com/v3/__https://github.com/scottedwardpratt__;!!HXCxUKc!10Dm_i8a1QmXyxD1If94ZGFTN60rZMdQVOOsXs9TzhsKjtfz6ON5gHVqa0kdVAg69lDzPBDqq3l7w9lSfZFT-h4$> — Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/bandframework/bandframework/pull/131*pullrequestreview-2326841388__;Iw!!HXCxUKc!10Dm_i8a1QmXyxD1If94ZGFTN60rZMdQVOOsXs9TzhsKjtfz6ON5gHVqa0kdVAg69lDzPBDqq3l7w9lS7Ae740g$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AB6P62WFNP3E4CBERN6UWYLZYIOCZAVCNFSM6AAAAABOOAHBRGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGMRWHA2DCMZYHA__;!!HXCxUKc!10Dm_i8a1QmXyxD1If94ZGFTN60rZMdQVOOsXs9TzhsKjtfz6ON5gHVqa0kdVAg69lDzPBDqq3l7w9lS19nNY94$>. You are receiving this because you were mentioned.Message ID: ***@***.***>

Great, I deleted the jpg over at https://github.com/bandframework/bandframework/commit/980c23feb0ca9fee38a2d280de258df4f22ab537
Thanks

wildsm committed:
Update .wordlist.txt

Attempting to remove gKTCLEAGE

ozgesurer committed:
Updating the links for QGP_Bayes and nsat

asemposki committed:
updating submodule SAMBA

wildsm committed:
Merge pull request #139 from asemposki/develop

updating submodule SAMBA SDK version

wildsm committed:
Update AUTHORS

Added SmoothEmulator authors

wildsm committed:
Update README.md

Added SmoothEmulator authors

wildsm committed:
Update bandsdk.md

Added SmoothEmulator authors

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update CHANGELOG.rst

Corrected two links in CHANGELOG (because it is rst not md)

ozgesurer committed:
Update README.md

minor change in readme

ozgesurer committed:
Merge pull request #138 from bandframework/v04/documentation

Documentation associated with the v0.4 release

ozgesurer committed:
Update README.md

adding periods

ozgesurer commented:
I have reviewed the develop branch and confirmed that all links are working properly. I also checked the SDK links and verified the numbers. Additionally, I created one more PR to add two periods to the README. @wildsm, after merging the documentation branch into develop, a reviewer is required for the changes.

I believe the only remaining task is to update the version number wherever it appears and remove the +dev suffix before merging into the main. Let me know if you’d like me to handle this.

ozgesurer committed:
Merge pull request #140 from bandframework/v04/documentation

Update README.md

wildsm commented:
> I have reviewed the develop branch and confirmed that all links are working properly. I also checked the SDK links and verified the numbers. Additionally, I created one more PR to add two periods to the README. @wildsm, after merging the documentation branch into develop, a reviewer is required for the changes.
> 
> I believe the only remaining task is to update the version number wherever it appears and remove the +dev suffix before merging into the main. Let me know if you’d like me to handle this.

@ozgesurer - Thanks! To confirm: the 'one more PR' has already been merged into `develop`, correct?

ozgesurer commented:
> > I have reviewed the develop branch and confirmed that all links are working properly. I also checked the SDK links and verified the numbers. Additionally, I created one more PR to add two periods to the README. @wildsm, after merging the documentation branch into develop, a reviewer is required for the changes.
> > I believe the only remaining task is to update the version number wherever it appears and remove the +dev suffix before merging into the main. Let me know if you’d like me to handle this.
> 
> @ozgesurer - Thanks! To confirm: the 'one more PR' has already been merged into `develop`, correct?

Yes, Moses approved it.

wildsm committed:
Update README.md

wildsm committed:
Dropping +dev in README

DanielRPhillips committed:
Update README.md

Changed description of BAND examples

DanielRPhillips committed:
Update README.md

Re-factored tools & examples to try and align with new definition thereof.

DanielRPhillips committed:
Update README.md

Fixed typo; re-ordered tools according to BAND workflow; re-ordered examples alphabetically

DanielRPhillips committed:
Merge pull request #141 from bandframework/v04/documentation

Documentation and overall changes for v0.4

wildsm committed:
Update README.md

wildsm committed:
Update CHANGELOG.rst

Changing release date

DanielRPhillips committed:
Merge pull request #145 from bandframework/wildsm-patch-1

Update README.md

wildsm committed:
Update CHANGELOG.rst

DanielRPhillips committed:
Merge pull request #146 from bandframework/v04/dailyupdatingofdate

Update CHANGELOG.rst

wildsm committed:
Update bandsdk.md

DanielRPhillips reviewed (approved):
I have reviewed the top level changes to bandframework and approve the merge.
